multi-vm-chef-template-ubuntu-vm/azuredeploy.json

```diff
@@ -1,291 +1,312 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-  "contentVersion": "1.0.0.0",
-  "parameters": {
-    "newStorageAccountName": {
-      "type": "string",
-      "metadata": {
-        "description": "Unique  Name for the Storage Account where the Virtual Machine's disks will be placed"
-      }
-    },
-    "vmDnsName": {
-      "type": "string",
-      "metadata": {
-        "description": "DNS  name for the VM"
-      }
-    },
-    "numberOfInstances": {
-      "type": "int",
-      "defaultValue": 2,
-      "metadata": {
-        "description": "Number of VM instances to create, default is 2"
-      }
-    },
-    "availabilitySetName": {
-      "type": "string",
-      "metadata": {
-        "description": "Availability Set name for the VMs"
-      }
-    },
-    "adminUserName": {
-      "type": "string",
-      "metadata": {
-        "description": "Admin user name for the Virtual Machines"
-      }
-    },
-    "adminPassword": {
-      "type": "securestring",
-      "metadata": {
-        "description": "Admin password for the Virtual Machines"
-      }
-    },
-    "imagePublisher": {
-      "type": "string",
-      "defaultValue": "Canonical",
-      "metadata": {
-        "description": " Publisher for the OS image, the default is Canonical"
-      }
-    },
-    "imageOffer": {
-      "type": "string",
-      "defaultValue": "UbuntuServer",
-      "metadata": {
-        "description": "The name of the image offer. The default is Ubuntu"
-      }
-    },
-    "imageSKU": {
-      "type": "string",
-      "defaultValue": "14.04.2-LTS",
-      "metadata": {
-        "description": "Version of the image. The default is 14.04.2-LTS"
-      }
-    },
-    "vmSize": {
-      "type": "string",
-      "defaultValue": "Standard_A0",
-      "metadata": {
-        "description": "VM Size"
-      }
-    },
-    "chef_node_name": {
-      "type": "string",
-      "metadata": {
-        "description": "Unique name for the node (VM) in the Chef Organization"
-      }
-    },
-    "chef_server_url": {
-      "type": "string",
-      "metadata": {
-        "description": "Organization URL for the Chef Server. Example https://ChefServerDnsName.cloudapp.net/organizations/Orgname"
-      }
-    },
-    "validation_client_name": {
-      "type": "string",
-      "metadata": {
-        "description": "Validator key name for the organization. Example : MyOrg-validator"
-      }
-    },
-    "runlist": {
-      "type": "string",
-      "defaultValue": "recipe[getting-started]",
-      "metadata": {
-        "description": "Optional Run List to Execute"
-      }
-    },
-    "validation_key": {
```

Comment:
`validation_key` is going to differ for every user and hence it makes sense to pass it as a parameter.

Beauty of these JSON templates is that, all the parameters and variables can be replaced by the hardcoded values, and the template will still work. But that's not a good practice.